packages/@aws-containers/aws-ecs-builder/lib/extensions/appmesh.ts

```diff
@@ -0,0 +1,216 @@
+        interval: cdk.Duration.seconds(5),
+        timeout: cdk.Duration.seconds(2),
+      },
+      memoryReservationMiB: 128,
+      user: '1337',
+      logging: new ecs.AwsLogDriver({ streamPrefix: 'envoy' }),
+    });
+
+    // Raise the number of open file descriptors allowed.
+    this.container.addUlimits({
+      softLimit: 1024000,
+      hardLimit: 1024000,
+      name: ecs.UlimitName.NOFILE,
+    });
+  }
+
+  // Enable cloudmap for the service
+  public mutateServiceProps(props: ServiceBuild) {
+    return {
+      ...props,
+
+      // We must make sure that service tasks are registered into
+      // CloudMap so that the App Mesh proxy can find them.
+      cloudMapOptions: {
+        dnsRecordType: 'A',
+        dnsTtl: cdk.Duration.seconds(10),
+        failureThreshold: 2,
+        name: this.parentService.id,
+      },
+
+      // These specific deployment settings are currently required in order to
+      // maintain availability during a rolling deploy of the service with App Mesh
+      // Warning, these settings do not work with a low task count however.
+      // minHealthyPercent: 100,
+      // maxHealthyPercent: 125,
+    } as ServiceBuild;
+  }
+
+  // Now that the service is defined we can create the AppMesh virtual service
+  // and virtual node for the real service
+  public useService(service: ecs.Ec2Service | ecs.FargateService) {
+    const containerextension = this.parentService.serviceDescription.get('service-container') as Container;
+
+    if (!containerextension) {
+      throw new Error('Firelens extension requires an application extension');
+    }
+
+    const cloudmapNamespace = this.parentService.cluster.defaultCloudMapNamespace;
+
+    if (!cloudmapNamespace) {
+      throw new Error('You must add a CloudMap namespace to the ECS cluster in order to use the AppMesh extension');
+    }
+
+    // Create a virtual node for the name service
+    this.virtualNode = new appmesh.VirtualNode(this.scope, `${this.parentService.id}-virtual-node`, {
+      mesh: this.mesh,
+      virtualNodeName: this.parentService.id,
+      cloudMapService: service.cloudMapService,
+      listener: {
+        portMapping: {
+          port: containerextension.trafficPort,
+          protocol: appmesh.Protocol.HTTP,
+        },
+        // Virtual node health is disabled, as we already have a Cloudmap healthcheck
+        // and container healthcheck from ECS
+        /*healthCheck: {
+          healthyThreshold: 2,
+          intervalMillis: 5000, // minimum
+          path: '/',
+          port: this.portNumber,
+          protocol: appmesh.Protocol.HTTP,
+          timeoutMillis: 2000, // minimum
+          unhealthyThreshold: 2
+        }*/
+      },
+    });
+
+    // Create virtual service to make the virtual node accessible
+    this.virtualService = new appmesh.VirtualService(this.scope, `${this.parentService.id}-virtual-service`, {
+      mesh: this.mesh,
+      virtualNode: this.virtualNode,
```

Comment:
I think in the future we'll want to put a Virtual Router in here instead (even if only one target), but we can make that change later.

Reply:
Fixed!